src/engine/automaton/automaton.ts

```diff
@@ -93,6 +93,105 @@ export class Automaton implements IAutomaton {
+      for (const rightSymbol in this.deltaFunctionMatrix[leftSymbol]) {
+        const edges = this.deltaFunctionMatrix[leftSymbol][rightSymbol];
+        for (const edge of edges) {
+          alphabet.add(edge.inputChar);
+          if (this.automatonType == AutomatonType.PDA && edge instanceof PDAEdge) {
+            stackAlphabet.add(edge.readStackChar);
+            for (const symbol of edge.writeStackWord) {
+              stackAlphabet.add(symbol);
+            }
+          }
+        }
+      }
+    }
+
+    switch (this.automatonType) {
+      case AutomatonType.FINITE:
+        // if there is Epsilon it is non-deterministic
+        if (alphabet.has(EPSILON)) {
+          return false;
+        }
+        // if any state has no edges in deltafunction, automaton is non-deterministic
+        if (this.states.length != Object.keys(this.deltaFunctionMatrix).length) {
+          return false;
+        }
+        // now, for each node we go through the deltaMatrix a second time and for each edge we check
+        // if all the symbols are present and if the number of edges
+        // equals this number, therefore we check if each edge is used only once
+        for (const leftSymbol in this.deltaFunctionMatrix) {
+          const found = new Set<string> ([]);
+          let edgeNum = 0;
+
+          for (const rightSymbol in this.deltaFunctionMatrix[leftSymbol]) {
+            const edges = this.deltaFunctionMatrix[leftSymbol][rightSymbol];
+            for (const edge of edges) {
+              found.add(edge.inputChar);
+              edgeNum++;
+            }
+          }
+          if (edgeNum != found.size || edgeNum != alphabet.size) {
+            return false;
+          }
+        }
+        return true;
+
+      case AutomatonType.PDA:
+        // if there is Epsilon it is non-deterministic
+        if (alphabet.has(EPSILON)) {
+          return false;
+        }
+        // if any state has no edges in deltafunction, automaton is non-deterministic
+        if (this.states.length != Object.keys(this.deltaFunctionMatrix).length) {
+          return false;
+        }
+        // Same as before only now we compare the number of unique alpha-stack combinations and actual edges
+        // and number of actual edges to all combinations (alphabet.size x stackAlphabet.size)
+        for (const leftSymbol in this.deltaFunctionMatrix) {
+          const found: Record<string, Set<string>> = {};
+          let edgeNum = 0;
+
+          for (const rightSymbol in this.deltaFunctionMatrix[leftSymbol]) {
+            const edges = this.deltaFunctionMatrix[leftSymbol][rightSymbol];
+            for (const edge of edges) {
+              if (edge instanceof PDAEdge) {
+                if (found[edge.inputChar] === undefined) {
+                  found[edge.inputChar] = new Set<string> ([]);
+                }
+                found[edge.inputChar].add(edge.readStackChar);
+                edgeNum++;
```

Comment:
teoreticky toto nemusíš počítať kúsok po kúsku, iba pričítaš dĺźku arrayu:
```suggestion
            const edges = this.deltaFunctionMatrix[leftSymbol][rightSymbol];
            edgeNum += edges.length
            for (const edge of edges) {
              if (edge instanceof PDAEdge) {
                if (found[edge.inputChar] === undefined) {
                  found[edge.inputChar] = new Set<string> ([]);
                }
                found[edge.inputChar].add(edge.readStackChar);
                edgeNum++;
```

Reply:
Jasné, dal som to tam.